hooks/useCart/index.ts

```diff
@@ -1,22 +1,69 @@
 import { AddItem } from "@spree/storefront-api-v2-sdk/types/interfaces/endpoints/CartClass";
-import { IToken } from "@spree/storefront-api-v2-sdk/types/interfaces/Token";
+import { IOrder } from "@spree/storefront-api-v2-sdk/types/interfaces/Order";
 import { useQuery } from "react-query";
 import { spreeClient } from "../../config/spree";
+import { QueryKeys } from "../queryKeys";
 
-const createCart = async (token: IToken) => {
-  const response = await spreeClient.cart.create(token);
-  if (response.isSuccess()) {
-    return response.success();
+export const showCart = async () => {
+  const storage = (await import("../../config/storage")).default;
+  const token = await storage.getToken();
+  if (token) {
+    const getCart = await spreeClient.cart.show({ bearerToken: token.access_token });
+    if (getCart.isSuccess()) {
+      return getCart.success();
+    } else {
+      const newCart = await spreeClient.cart.create({ bearerToken: token.access_token });
+      if (newCart.isSuccess()) {
+        return newCart.success();
+      } else {
+        throw new Error(newCart.fail().message);
+      }
+    }
+  } else {
+    const orderToken = storage.getGuestOrderToken();
+    if (orderToken) {
+      const response = await spreeClient.cart.show(orderToken);
+      if (response.isSuccess()) {
+        return response.success();
+      } else {
+        throw new Error(response.fail().message);
+      }
+    } else {
+      const response = await spreeClient.cart.create();
+      if (response.isSuccess()) {
+        const result = response.success();
+        storage.setGuestOrderToken({ orderToken: result.data.attributes.token });
+        return result;
+      } else {
+        throw new Error(response.fail().message);
+      }
+    }
   }
 };
 
-const showCart = async (token: IToken) => {
-  const response = await spreeClient.cart.show(token);
+export const addItemToCart = async (item: AddItem) => {
+  const storage = (await import("../../config/storage")).default;
+  const token = await storage.getToken();
+  if (!token) {
+    const orderToken = storage.getGuestOrderToken();
+    if (orderToken) {
+      const response = await spreeClient.cart.addItem(orderToken, item);
+      if (response.isSuccess()) {
+        return response.success();
+      } else {
+        throw new Error(response.fail().message);
+      }
+    }
+    throw new Error("NO CART TOKENS FOUND, COULD NOT ADD ITEM");
```

Comment:
I was getting this after logging in... but I think the login function is still broken

Reply:
After starting up `dna-admin` locally and connecting the frontend to work, login now works, but "add to cart" returns this:
```
xhr.js?b50d:177 POST http://localhost:8080/api/v2/storefront/cart/add_item?variant_id=165&quantity=1 404 (Not Found)
```

...after a page refresh, "add to cart" worked, curiously.